crates/sui-protocol-config/src/lib.rs

```diff
@@ -465,6 +466,10 @@ pub struct ProtocolConfig {
     /// 3f+1 must vote), while 0bps would indicate that 2f+1 is sufficient.
     buffer_stake_for_protocol_upgrade_bps: Option<u64>,
 
+    /// What version of reordering algorithm we are using for user transactions included in a Narwhal
+    /// commit in consensus handler.
+    consensus_user_transaction_reordering_version: Option<u64>,
```

Comment:
I thought this is similar to gas model in the sense that we will probably iterate a few times so I added it as a version number instead of feature flag.

Reply:
we could still use a feature flag, but make it an enum instead of a bool - i think that will help readability, since the enum variants can be descriptive.

Reply:
Good idea.